py/desispec/scripts/daily_processing.py

```diff
@@ -0,0 +1,239 @@
+                                        science_joint_fit, define_and_assign_dependency, create_and_submit, \
+                                        update_and_recurvsively_submit, checkfor_and_submit_joint_job
+from desispec.workflow.queue import update_from_queue, continue_looping
+
+
+def daily_processing_manager(specprod=None, exp_table_path=None, proc_table_path=None, path_to_data=None, procobstypes=None, camword=None,
+                             dry_run=False, tab_filetype='csv',override_night=None, queue='realtime'):
+    """
+    Generates processing tables for the nights requested. Requires exposure tables to exist on disk.
+
+    Args:
+        specprod: str. The name of the current production. If used, this will overwrite the SPECPROD environment variable.
+        exp_table_path: str. Full path to where to exposure tables are stored, WITHOUT the monthly directory included.
+        proc_table_path: str. Full path to where to processing tables to be written.
+        path_to_data: str. Path to the raw data.
+        procobstypes: str or comma separated list of strings. The exposure OBSTYPE's that you want to include in the processing table.
+        camword: str. Camword that, if set, overwrites the list of cameras found in the files and only runs on those given/
+                      Examples: a0123456789, a1, a2b3r3, a2b3r4z3.
+        dry_run: boolean. If true, no scripts are written and no scripts are submitted. The tables are still generated and written, however.
+        tab_filetype: str. The file extension (without the '.') of the exposure and processing tables.
+        override_night: str or int. 8 digit night, e.g. 20200314, of data to run on. If None, it runs on the current night.
+        queue: str. The name of the queue to submit the jobs to. Default is "realtime".
+
+    Returns: Nothing
+
+    Notes:
+        Generates both exposure table and processing tables 'on the fly' and saves them at various checkpoints. These
+        should be capable of being reloaded in case of interuption or accidental termination of the manager's process.
+    """
+    ## If not being done during operating hours, and we're not simulating data or running a catchup run, exit.
+    if not during_operating_hours(dry_run=dry_run) and override_night is None:
+        print("Not during operating hours, and not asked to perform a dry run or run on historic data. Exiting.")
+        sys.exit(0)
+
+    ## What night are we running on?
+    true_night = what_night_is_it()
+    if override_night is not None:
+        night = int(override_night)
+        print(f"True night is {true_night}, but running for night={night}")
+    else:
+        night = true_night
+
+    ## Get default values for input variables
+    if procobstypes is None:
+        procobstypes = default_exptypes_for_proctable()
+    if camword is not None:
+        print(f"Over-riding camword in data with user provided value: {camword}")
```

Comment:
let's convert `print` statements to `desiutil.log.get_logger` logging with info, debug, warning, etc.

Reply:
Okay. I followed the precedent of desi_dailyproc where print statements were used in the manager and the underlying code was the only ones using the official logger. But I will make that change

Reply:
You are good at catching me requesting a change to something that you copied from me :)

Retroactively justifying my inconsistency: when all the outputs come from a single script and they occur in linear order, I like "print" for the extra readability of not having all the prefix stuff.  But once code gets spread over multiple functions in multiple modules, and admittedly when I am less familiar with the code, I find it useful to have the extra module names and line numbers to assist with understanding where a message is coming from, even if it is uglier to read.  And especially if the code starts supporting verbose/non-verbose/quiet stuff, different log levels start looking more attractive.

I think I was also confused by messages like
```
Outfile is: 
[] <class 'numpy.ndarray'>
```
and
```
Starting iteration 1
```
and had a knee-jerk reaction to want module names and line numbers.  It might still be ok to have readable prints at the very top level script, and only underlying modules use logging.

Reply:
Cleaned up some debug print statements that aren't needed and changed all the print statements under the /workflow/ subdir to be desi logger calls. I retain the print statements in the scripts themselves for what I think is better readability, but I don't feel strongly about that and could be convinced to change those as well.